wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Addresses.hs

```diff
@@ -51,10 +60,64 @@ newAddress NewAddress {..} = do
               >>= migrate
 
 -- | Verifies that an address is base58 decodable.
-verifyAddress :: Monad m => Text -> m (WalletResponse AddressValidity)
-verifyAddress address =
-    case decodeTextAddress address of
-        Right _ ->
-            return $ single $ AddressValidity True
-        Left _  ->
-            return $ single $ AddressValidity False
+getAddress
+    :: (MonadThrow m , V0.MonadWalletLogic ctx m)
+    => Text
+    -> m (WalletResponse WalletAddress)
+getAddress addrText = do
+    addr <- either
+        (throwM . InvalidAddressFormat)
+        pure
+        (decodeTextAddress addrText)
+
+    ss <- V0.askWalletSnapshot
+
+    let
+        addrId :: V0.CId V0.Addr
+        addrId =
+            V0.addressToCId addr
```

Comment:
I'm not sure this is right. `CId` is a newtype around a hash with a phantom type.